components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/test/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrantTest.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    @Test(expectedExceptions = IdentityOAuth2ClientException.class)
+    public void testWhenSwitchingOrgIsInvalid() throws IdentityOAuth2Exception, OrganizationManagementServerException {
+
+        when(mockOAuth2TokenValidationResponseDTO.isValid()).thenReturn(true);
+        when(mockAccessTokenDO.getAuthzUser()).thenReturn(mockAuthenticatedUser);
+        when(mockAuthenticatedUser.getTenantDomain()).thenReturn(TOKEN_ISSUED_TENANT_DOMAIN);
+        when(mockOrganizationManager.getAncestorOrganizationIds(nullable(String.class))).thenReturn(new ArrayList<>());
+        organizationSwitchGrant.validateGrant(mockOAuthTokenReqMessageContext);
+    }
+
+    @Test(expectedExceptions = IdentityOAuth2ClientException.class)
+    public void testSwitchDifferentSubTree() throws IdentityOAuth2Exception, OrganizationManagementException {
+
+        when(mockOAuth2TokenValidationResponseDTO.isValid()).thenReturn(true);
+        when(mockAccessTokenDO.getAuthzUser()).thenReturn(mockAuthenticatedUser);
+        when(mockAuthenticatedUser.getTenantDomain()).thenReturn(TOKEN_ISSUED_TENANT_DOMAIN);
+        when(mockOrganizationManager.resolveOrganizationId(anyString())).thenReturn(TOKEN_ISSUED_ORG_ID);
+        when(mockOrganizationManager.getAncestorOrganizationIds(nullable(String.class))).thenReturn(
+                Arrays.asList("123", "456"));
+        organizationSwitchGrant.validateGrant(mockOAuthTokenReqMessageContext);
+    }
+
+    @Test(expectedExceptions = IdentityOAuth2ClientException.class)
+    public void testSwitchSameOrganization() throws IdentityOAuth2Exception, OrganizationManagementException {
+
+        when(mockOAuth2TokenValidationResponseDTO.isValid()).thenReturn(true);
+        when(mockAccessTokenDO.getAuthzUser()).thenReturn(mockAuthenticatedUser);
+        when(mockAuthenticatedUser.getTenantDomain()).thenReturn(TOKEN_ISSUED_TENANT_DOMAIN);
+        when(mockOrganizationManager.resolveOrganizationId(anyString())).thenReturn(TOKEN_ISSUED_ORG_ID);
+        organizationSwitchGrant.validateGrant(mockOAuthTokenReqMessageContext);
+    }
+
+    @Test
+    public void testSuccessfullySwitchToken()
+            throws OrganizationManagementException, IdentityOAuth2Exception, UserIdNotFoundException {
+
+        when(mockOAuth2TokenValidationResponseDTO.isValid()).thenReturn(true);
+        when(mockAccessTokenDO.getAuthzUser()).thenReturn(mockAuthenticatedUser);
+        when(mockAuthenticatedUser.getTenantDomain()).thenReturn(TOKEN_ISSUED_TENANT_DOMAIN);
+        when(mockOrganizationManager.resolveOrganizationId(anyString())).thenReturn(SWITCHING_ORG_ID);
+        when(mockOrganizationManager.getAncestorOrganizationIds(nullable(String.class))).thenReturn(
+                Arrays.asList("12345", SWITCHING_ORG_ID));
+        when(mockAuthenticatedUser.getUserId()).thenReturn("12345");
+        organizationSwitchGrant.validateGrant(mockOAuthTokenReqMessageContext);
+    }
```

Comment:
add a test case to check binding is set properly.

Reply:
Added by - https://github.com/wso2-extensions/identity-oauth2-grant-organization-switch/pull/14/commits/c5813e8893b0c0f0e0cdad0d6903ebbcc53ec4a1